Dockerfile

```diff
@@ -20,29 +11,36 @@ RUN apt-get install -qqqy \
     poppler-utils \
     build-essential \
     pkg-config \
-    locales
+    locales \
+  && apt-get -q clean
 
+# Build locales to avoid encoding issues with Scrapy encoding
 RUN locale-gen en_GB.UTF-8
 
 ENV LC_ALL=en_GB.UTF-8
 ENV LANG=en_GB.UTF-8
 ENV LANGUAGE=en_GB.UTF-8
 
-# Update pip
-RUN pip3 install -qqq --upgrade pip
-RUN rm -f /usr/bin/python && ln /usr/bin/python3 /usr/bin/python
-# Install any needed packages specified in requirements.txt
-RUN pip3 install -qqq -r requirements.txt
-RUN pip3 install -qqq scrapyd scrapyd-client
+# Set the working directory to /wsf_scraper
```

Comment:
Comments like this, which just explain *what* the code are doing, are rarely helpful (unless the code is very obtuse or complicated – but then it might be worth rewriting!).

Comments should explain *why* the code is written this way. What’s the requirement? Why have we chosen this style? What higher-order concept does this map to? etc.

The comment above – _Build locales to avoid encoding issues with Scrapy encoding_ – is better, because it explains the problems that would arise if we didn’t have that code.